src/main/java/com/zendesk/maxwell/producer/InflightMessageList.java

```diff
@@ -35,7 +35,10 @@ public synchronized void addMessage(BinlogPosition p) {
 	/* returns the position that stuff is complete up to, or null if there were no changes */
 	public synchronized BinlogPosition completeMessage(BinlogPosition p) {
 		InflightMessage m = this.linkedMap.get(p.toString());
-		assert(m != null);
+
+		if (m == null) {
+			return null;
+		}
```

Comment:
This shouldn't happen after this patch, right? In which case I'd probably leave this change out so we notice similar issues in the future.